Sources/SwiftGRPCNIO/ClientCalls/BaseClientCall.swift

```diff
@@ -233,7 +233,9 @@ extension BaseClientCall {
     if timeout == .infinite { return }
 
     self.connection.channel.eventLoop.scheduleTask(in: timeout.asNIOTimeAmount) { [weak self] in
```

Comment:
Would you mind adding timeout support to the server, too? Not in this PR, of course. When adding that, we should also make sure that _all_ promises (status and response) are being fulfilled (probably with an error) when the timeout occurs.

It might also be nice to set a maximum timeout on the server object, to prevent malicious clients from opening many requests with very long timeouts.

Reply:
We should definitely add this. I think we have higher priorities at the moment but I created #462 so we don't lose it!